### PR TITLE
Remove compliance check from the code

### LINKED ISSUE #114: Feature of exclude public channels seems to be breaking even after implementing changes in compliance export in Mattermost.
@fmartingr While testing the master branch for this plugin, I found an issue regarding the changes in the PR #102 .
The functionality added of excluding public channels is not working and the user receives an error on disabling and saving it. Also, since there is an extra check to validate that the compliance export is enabled, even after enabling the same, the user receives the same error.
I am providing the screenshots for the same:-
image

image

Logs:-
{"timestamp":"2024-09-24 20:14:01.170 +05:30","level":"error","msg":"Compliance monitoring must be enabled on the Mattermost server in order to include public channels in a LegalHold","caller":"app/plugin_api.go:1007","plugin_id":"com.mattermost.plugin-legal-hold"}

This shows that the compliance export needs to be enabled, which already is, still user receives this error

### PR DESCRIPTION
**Pull request by fmartingr**
#### Summary
- Removes compliance check from the code.
- Refactors the `Exclude public channels` check to `Include public channels` (defaults to `false`).
- Shows a small disclaimer if `Include public channels` is checked.
- Shows underlying errors in the update form, rather than one generic error.
- Fixes a bug where the state of the current active load test was not "reset", causing errors on the update form.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-59830

#### Screenshots
Screenshot 2024-09-25 at 18 28 55

Replaces #115 
Closes #114

### CODE REVIEW

File: webapp/src/components/create_legal_hold_form.tsx [hunk 1]
```diff
@@ -152,15 +153,19 @@ const CreateLegalHoldForm = (props: CreateLegalHoldFormProps) => {
...
                         />
-                        <label htmlFor={'legal-hold-exclude-public-channels'}>
-                            {'Exclude public channels'}
+                        <label htmlFor={'legal-hold-include-public-channels'}>
+                            {'Include public channels'}
                         </label>
                     </div>
+                    <div style={{display: (includePublicChannels) ? 'block' : 'none'}}>
+                        <i className='icon icon-alert-outline'/>
+                        <span>{'It is possible for users to access public content without becoming members of a public channel. This setting only captures public channels the users are members of.'}</span>
```
Comment (wiggin77) [inline webapp/src/components/create_legal_hold_form.tsx:167]:
```suggestion
                        <span>{'It is possible for users to access public content without becoming members of a public channel. This setting only captures public channels that users are members of.'}</span>
```

File: webapp/src/components/create_legal_hold_form.tsx [hunk 2]
```diff
@@ -152,15 +153,19 @@ const CreateLegalHoldForm = (props: CreateLegalHoldFormProps) => {
...
                         />
-                        <label htmlFor={'legal-hold-exclude-public-channels'}>
-                            {'Exclude public channels'}
+                        <label htmlFor={'legal-hold-include-public-channels'}>
+                            {'Include public channels'}
                         </label>
                     </div>
+                    <div style={{display: (includePublicChannels) ? 'block' : 'none'}}>
+                        <i className='icon icon-alert-outline'/>
+                        <span>{'It is possible for users to access public content without becoming members of a public channel. This setting only captures public channels that users are members of.'}</span>
```
Comment (wiggin77) [inline webapp/src/components/create_legal_hold_form.tsx:167]: @cwarnermm can we capture this in the documentation as well?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1: :+1:

File: webapp/src/components/create_legal_hold_form.tsx (hunk 2 above)
Comment (cwarnermm) [reply webapp/src/components/create_legal_hold_form.tsx:167]: https://github.com/mattermost/docs/pull/7411 updated accordingly

**Review by esarafianou — APPROVED** [review]

**Comment by fmartingr** [comment]
@AayushChaudhary0001 This PR is ready for QA.

**Comment by AayushChaudhary0001** [comment]
I was testing this PR and had a suggestion on this.
The helper text added for the functionality looks a bit confusing, can you please change it and add what this functionality actually is responsible for after enabling and disabling.
image

**Comment by AayushChaudhary0001** [comment]
On testing this PR, I am facing an error, inspite of changing the compliance monitoring to true or false. Earlier it was due to compliance monitoring but I don't think this time its the same reason. Please refer the screenshot below.
image

**Comment by AayushChaudhary0001** [comment]
I am facing an other issue in this PR, the changes done in any legal hold are not getting saved in a single time, and some fields are changed on opening the legal hold creation modal again. Please refer the screenshots below. FYI, no new change is found in the logs.

Existing legal hold
image

On updating and saving
image

Again updating, when opened the the modal
image

**Comment by fmartingr** [comment]
Hey @AayushChaudhary0001, thanks for catching that. I just pushed a fix, I'm not experiencing that behavior anymore:

https://github.com/user-attachments/assets/c5d32e80-0ff4-4b9e-8585-a721c883b4c7

I also fixed the Update/Show secret tooltips that I saw they were switched out.

**Comment by fmartingr** [comment]
Hey @AayushChaudhary0001 I just saw you 👍 my comment. Did you test that on your end again and it was fixed and QA passing?

**Comment by AayushChaudhary0001** [comment]
@fmartingr Will let you know soon, currently under QA testing.

**Comment by fmartingr** [comment]
> @fmartingr The fix has been tested and works well. The compliance check has also been removed, user is able to include and exclude the public channels without any dependency of compliance monitoring. But the size of ZIP file getting downloaded on enabling or disabling the including public channels functionality seems to be similar, can you please take a look at it?

Hey @AayushChaudhary0001, I just checked two legal holds for the same users and same dates and one of them got the public channels and the other didn't. Did you check the content of the files or the output with the processor?

**Comment by AayushChaudhary0001** [comment]
@fmartingr On editing the config, I am facing this error, Can you please have a look at it why is this happening?
I am attaching a screenshot and server logs as well.

image


{"timestamp":"2024-10-03 18:25:59.942 +05:30","level":"error","msg":"could not update legal hold as it has already been updated by someone else","caller":"app/plugin_api.go:1003","plugin_id":"com.mattermost.plugin-legal-hold"}

I am testing this on my local instance and there is no other user present here.

**Comment by AayushChaudhary0001** [comment]
The above issue is not now reproducible

**Review by AayushChaudhary0001 — COMMENTED** [review]
This PR has been tested for the compliance check removed, due to which the user is able to include or exclude the public channels from legal hold without any dependency. These changes are working well, LGTM. Approved.

**Pull request merged**